PictureApp.API/Controllers/FollowersController.cs

```diff
@@ -0,0 +1,76 @@
+using System;
+using System.Security.Authentication;
+using System.Security.Claims;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.Mvc;
+using PictureApp.API.Extensions.Exceptions;
+using PictureApp.API.Services;
+
+namespace PictureApp.API.Controllers
+{
+    [Authorize]
+    [Route("api/users/{userId}/followers")]
+    [ApiController]
+    public class FollowersController : ControllerBase
+    {
+        private readonly IUserService userService;
+        private readonly IFollowerService _followerService;
+
+        public FollowersController(IUserService userService, IFollowerService followerService)
+        {
+            this.userService = userService;
+            _followerService = followerService;
+        }
+
+        [HttpPost("{id}/setfollow")]
+        public async Task<IActionResult> SetUpFollower(int userId, int id)
+        {
+            if (userId != int.Parse(User.FindFirst(ClaimTypes.NameIdentifier).Value))
```

Comment:
In this case we do authorization check twice. First time here and the second one in _followerService.SetUpFollower.

Reply:
fixed by:
https://github.com/anowakowski/PictureApp_API/pull/29/commits/75558ef8a39ce9d95f1a826d45e2793b5e1a8440